pkg/policy/validate_test.go

```diff
@@ -1293,39 +1293,24 @@ func Test_checkAutoGenRules(t *testing.T) {
 		expectedResult bool
```

Comment:
The tests added in the [previous](https://github.com/kyverno/kyverno/pull/1839) PR are duplicated, updated to cover other scenarios.